code/modules/awaymissions/mission_code/mining.dm

```diff
@@ -0,0 +1,398 @@
+	return
+
+/mob/living/simple_animal/hostile/megafauna/bubblegum/virtual
+	name = "bubblegum hologram"
+	desc = "A holographic version of the king of the slaughter demons. You feel something oddly real staring back at you."
+	crusher_loot = list()
+	loot = list()
+	virtual = 1
+	true_spawn = 0
+
+/mob/living/simple_animal/hostile/megafauna/bubblegum/virtual/Initialize()
+	. = ..()
+	qdel(internal)
+
+// need this otherwise bubbles can teleport out of his arena
+/mob/living/simple_animal/hostile/megafauna/bubblegum/virtual/hallucination_charge_around(var/times = 4, var/delay = 6, var/chargepast = 0, var/useoriginal = 1)
+	var/startingangle = rand(1, 360)
+	if(!target)
+		return
+	var/turf/chargeat = get_turf(target)
+	var/srcplaced = 0
+	for(var/i = 1 to times)
+		var/ang = (startingangle + 360/times * i)
+		if(!chargeat)
+			return
+		var/turf/place = locate(chargeat.x + cos(ang) * times, chargeat.y + sin(ang) * times, chargeat.z)
+		if(!place)
+			continue
+		if(!srcplaced && useoriginal && get_dist(nest.parent, place) <= MEGAFAUNA_NEST_RANGE)
+			forceMove(place)
+			srcplaced = 1
+			continue
+		var/mob/living/simple_animal/hostile/megafauna/bubblegum/hallucination/B = new /mob/living/simple_animal/hostile/megafauna/bubblegum/hallucination(src.loc)
+		B.forceMove(place)
+		INVOKE_ASYNC(B, .proc/charge, chargeat, delay, chargepast)
+	if(useoriginal)
+		charge(chargeat, delay, chargepast)
+
+/mob/living/simple_animal/hostile/megafauna/bubblegum/virtual/grant_achievement(medaltype, scoretype, crusher_kill)
+	return
+
+/mob/living/simple_animal/hostile/megafauna/colossus/virtual
+	name = "colossus hologram"
+	desc = "A holographic god. One of the strongest creatures that has ever lived."
+	crusher_loot = list()
+	loot = list()
+	virtual = 1
+
+/mob/living/simple_animal/hostile/megafauna/colossus/virtual/Initialize()
+	. = ..()
+	qdel(internal)
+
+/mob/living/simple_animal/hostile/megafauna/colossus/virtual/grant_achievement(medaltype, scoretype, crusher_kill)
+	return
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/virtual
+	name = "hierophant hologram"
+	desc = "A holographic club. It's said to wipe from existence those who fall to its rhythm."
+	loot = list()
+	crusher_loot = list()
+	virtual = 1
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/virtual/Initialize()
+	. = ..()
+	qdel(internal)
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/virtual/spawn_crusher_loot()
+	return
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/virtual/grant_achievement(medaltype, scoretype, crusher_kill)
+	return
+
+/mob/living/simple_animal/hostile/megafauna/legion/virtual
+	name = "Legion Hologram"
+	desc = "One of many... holograms."
+	loot = list()
+	virtual = 1
+
+/mob/living/simple_animal/hostile/megafauna/legion/virtual/Initialize()
+	. = ..()
+	qdel(internal)
+
+/mob/living/simple_animal/hostile/megafauna/legion/virtual/death()
```

Comment:
what parts need to be overwritten here compared to original legion besides the spawntype which we can insert into a variable